lib/fs.js

```diff
@@ -2023,22 +2023,25 @@ function ReadStream(path, options) {
     }
 
     this.pos = this.start;
-  }
-
-  // Backwards compatibility: Make sure `end` is a number regardless of `start`.
-  // TODO(addaleax): Make the above typecheck not depend on `start` instead.
-  // (That is a semver-major change).
-  if (typeof this.end !== 'number')
-    this.end = Infinity;
+  } else {
+    // Backwards compatibility: Make sure `end` is a number regardless of
+    // `start`.
+    // TODO(addaleax): Make the above typecheck not depend on `start` instead.
+    // (That is a semver-major change).
+    if (typeof this.end !== 'number')
+      this.end = Infinity;
+    else if (!Number.isInteger(this.end))
+      throw new ERR_INVALID_ARG_TYPE('end', 'integer', this.end);
 
-  if (typeof this.fd !== 'number')
```

Comment:
This should not be included in the `else` block. The if / else is only about `this.start` and `this.end` and nothing else.

Reply:
Yes, it should not.